src/config.rs

```diff
@@ -82,6 +84,18 @@ impl Config for FsConfig {
         let shell_path = self.root_path.join("shells").join(name);
         shell_path.is_dir()
     }
+
+    fn get_shell_list(&self) -> Vec<String> {
+        let mut shell_names = Vec::new();
+        let paths = fs::read_dir(self.shell_root_path().to_owned()).expect("read_dir failed in config get_shell_list function");
+        for path in paths {
+            let shell_path = path.unwrap().path();
+            let shell_name = shell_path.file_name().unwrap().to_str().unwrap().to_owned();
+            shell_names.push(shell_name);
+        }
+        shell_names.sort();
+        return shell_names;
```

Comment:
Need to do some error handling here as mentioned in slack.  This will probably mean propagating the errors to the caller via a `Result` of some kind, and possibly introducing a new error type.